agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,688 @@
+                             power_state.lower(), vm_name)
+
+        def power_cycle_vm(self, vm_name=None, vm_uuid=None, timeout=None):
+                """
+                Power cycle a VM
+                """
+                resp = None
+                status = None
+
+                if not timeout:
+                        timeout = MIN_TIMEOUT
+
+                if not vm_name and not vm_uuid:
+                        logging.error("Require at least one of VM name or VM UUID")
+                        raise AHVFenceAgentException("No arguments provided")
+
+                if not vm_uuid:
+                        vm_uuid = self._get_vm_uuid(vm_name)
+
+                resp = self._power_cycle_vm(vm_uuid)
+                task_id = resp.json()['data']['extId']
+
+                try:
+                        self._wait_for_task(task_id, timeout)
+                except AHVFenceAgentException as err:
+                        logging.error("Failed to power-cycle VM %s", vm_uuid)
+                        logging.error("VM power-cycle task failed with status, %s", status)
+                        raise AHVFenceAgentException from err
+
+                logging.info("Power-cycled VM, %s", vm_name)
+
+
+def connect(options):
+        """
+        Create a client instance and return it after verifying
+        that the client can connect.
+        """
+        host = options["--ip"]
+        username = options["--username"]
+        password = options["--password"]
+        verify_ssl = False
+
+        if "--ssl-secure" in options:
+                verify_ssl = True
+
+        client = V4NutanixClient(host, username, password, verify_ssl)
+
+        try:
+                client.list_vms(limit=1)
+        except AHVFenceAgentException as err:
+                logging.error("Connection to Prism Central Failed")
+                logging.error(err)
+                fencing.fail(fencing.EC_LOGIN_DENIED)
+
+        return client
+
+def list_vms(client, options):
+        """
+        List VMs
+        """
+
+        vm_list = None
+        limit = None
+        filter_str =  None
+
+        if "--filter" in options:
+                filter_str = options["--filter"]
+
+        if "--limit" in options:
+                limit = options["limit"]
+
+        try:
+                vm_list = client.list_vms(filter_str, limit)
+        except AHVFenceAgentException as err:
+                logging.error("Failed to list VMs")
+                logging.error(err)
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+        return vm_list
+
+def get_power_state(client, options):
+        """
+        Get power state of a VM
+
+        Args:
+            Client(V4NutanixClient instance): V4NutanixClient instance
+            options(dict): options dictionary
+
+        Returns:
+            None
+        """
+        vmid = None
+        name = None
+        power_state = None
+
+        if "--uuid" in options:
```

Comment:
I'm a little confused here. Like I asked above, does fence agent uses UUID or VM-name to perform ops?